internal/impl/messaging_service_impl.go

```diff
@@ -157,6 +158,46 @@ func (service *messagingServiceImpl) Connect() (ret error) {
 	return nil
 }
 
+func (service *messagingServiceImpl) UpdateProperty(property config.ServiceProperty, value interface{}) (ret error) {
+        // Verify that the value is of type string. This is necessary since the only values that are acceptable for the
+        // properties that are currently supported for modification are strings. We take an interface type as the parameter
+        // for the sake of forwards compatibility.
+        _, valueIsString := value.(string)
+        if !valueIsString {
+                return solace.NewError(&solace.InvalidDataTypeError{}, fmt.Sprintf(constants.UnableToModifyServicePropertyWithInvalidValue, property, value), nil)
+        }
+
+        var state = service.getState()
+        if !(state == messagingServiceStateConnected || state == messagingServiceStateNotConnected || state == messagingServiceStateConnecting) {
+                // If the service is not connected or connecting, then the service is not in a valid state for
+                // updating service properties, so we return an error.
+                return solace.NewError(&solace.IllegalStateError{}, fmt.Sprintf(constants.UnableToModifyPropertyOfDisconnectedService), nil)
+        }
+
+        // FFC: Currently the following array is only needed within this method, and this method is not on the data path,
+        // so we can tolerate allocating the array each time this method is called. This array could be moved elsewhere,
+        // but that would likely require the use of a global variable.
+        var modifiableProperties = [...]config.ServiceProperty {config.AuthenticationPropertySchemeOAuth2AccessToken, config.AuthenticationPropertySchemeOAuth2OIDCIDToken}
```

Comment:
This should be a constant defined outside the scope of the function.

Reply:
I made a global variable in the same file with the modifiable service properties so that it is now generally accessible by the API.